apps/roam/src/utils/initializeObserversAndListeners.ts

```diff
@@ -44,14 +44,24 @@ import {
   findBlockElementFromSelection,
 } from "~/utils/renderTextSelectionPopup";
 import { renderNodeTagPopupButton } from "./renderNodeTagPopup";
+import { formatHexColor } from "~/components/settings/DiscourseNodeCanvasSettings";
 
 let discourseNodes: DiscourseNode[] = [];
-let discourseTagSet: Set<string> = new Set();
+let discourseTagToStyle: Record<string, { color: string }> = {};
 
 const refreshDiscourseNodeCache = () => {
   discourseNodes = getDiscourseNodes();
-  discourseTagSet = new Set(
-    discourseNodes.flatMap((n) => (n.tag ? [n.tag.toLowerCase()] : [])),
+  discourseTagToStyle = discourseNodes.reduce(
```

Comment:
We should reuse the existing refreshConfigTree.  No need to re-invent the wheel 😀